astropy/wcs/src/wcslib_celprm_wrap.c

```diff
@@ -118,6 +121,8 @@ static void PyCelprm_dealloc(PyCelprm* self)
     if (self->prefcount && (--(*self->prefcount)) == 0) {
         free(self->x);
         free(self->prefcount);
+        self->x = NULL;
+        self->prefcount = NULL;
```

Comment:
It's actually failing at line 120, as if `celfree(self->x)` can not free the error message memory.

Reply:
Since I am also looking at `wcslib` source, I wonder if the error is linked to a missing `(*errp)->msg = NULL;` after [this line](https://github.com/astropy/astropy/blob/main/cextern/wcslib/C/wcserr.c#L106)

Reply:
I am no C++ expert, but `*errp` is freed right after, so there should not be any referenceable `(*errp)->msg`, should there?

https://github.com/astropy/astropy/blob/ae491a557560e3c1efd05aec578207205208f0a1/cextern/wcslib/C/wcserr.c#L105-L109

Reply:
Certainly can’t be referenced through *errp. I was thinking if the address had been taken before and saved to a local pointer then that pointer would now point to invalid memory. (Will say that that is a fairly unusual scenario but at least possible in C)

Easiest way to check might be to add that msg = NULL, recompile and check if the segfault disappears.

Reply:
(The segfault should change to NULL pointer deference error)